duckduckgo/Android
# Replace GlobalScope with AppCoroutineScope

Task/Issue URL: https://app.asana.com/0/488551667048375/1200220741299072/f
Tech Design URL: 
CC: 

**Description**:
This PR replaces all calls using `GlobalScope.launch()` with `AppCoroutineScope.launch()`.
Calls that weren't manually tested were unit tested.

**Steps to test this PR**:
1. On a fresh install, after initial onboarding, check that the animation around the fire icon shows correctly.
2. After tapping the fire icon, test that "Clear all tabs and data" and "Cancel" work correctly.
3. Long press the DuckDuckGo app icon, test that selecting "Clear data" works correctly.
4. Test that "Enable Privacy Protection" and "Disable Privacy Protection" work correctly.
5. Go to "Settings" > "Unprotected Sites". Test adding, editing and deleting of an unprotected site.
6. Test that tabs' preview images and favicons display correctly.
7. Start typing in the address bar. Test that tapping autocomplete entries and bookmark entries works correctly.
8. Go to "Settings" > "Share Feedback". Test that submitting positive, negative and broken site feedback works correctly.

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
